src/ess/reduce/time_of_flight/toa_to_tof.py

```diff
@@ -74,7 +50,11 @@ def compute_tof_lookup_table(
     simulation: SimulationResults,
     ltotal_range: LtotalRange,
     distance_resolution: DistanceResolution,
-    toa_resolution: TimeOfArrivalResolution,
+    time_resolution: TimeResolution,
+    pulse_period: PulsePeriod,
+    pulse_stride: PulseStride,
+    pulse_stride_offset: PulseStrideOffset,
+    error_threshold: LookupTableRelativeErrorThreshold,
 ) -> TimeOfFlightLookupTable:
```

Comment:
This function is so long that the probability that all bugs are caught in code review approaches zero. Can you split it up, so components can be reasoned about and tested individually? Maybe this should turn into a class, but free functions is also an option.

Reply:
Done.